### PR TITLE
chore(rocksdb): lower memory footprint

### PR DESCRIPTION
**Pull request by dpc**
As per https://github.com/facebook/rocksdb/wiki/Block-Cache#caching-index-filter-and-compression-dictionary-blocks:

> By default index, filter, and compression dictionary blocks (with the exception of the partitions of [partitioned indexes/filters](https://github.com/facebook/rocksdb/wiki/Partitioned-Index-Filters)) are cached outside of block cache, and users won't be able to control how much memory should be used to cache these blocks, other than setting max_open_files.

### CODE REVIEW

File: fedimint-rocksdb/src/lib.rs [hunk 1]
```diff
@@ -159,6 +161,28 @@ fn next_prefix(prefix: &[u8]) -> Option<Vec<u8>> {
 impl IRawDatabase for RocksDb {
     type Transaction<'a> = RocksDbTransaction<'a>;
     async fn begin_transaction<'a>(&'a self) -> RocksDbTransaction {
+        let count = self.1.fetch_add(1, std::sync::atomic::Ordering::Relaxed);
+        if count % 100_000 == 0 {
```
Comment (elsirion) [inline fedimint-rocksdb/src/lib.rs:165]: This is kinda hacky, do we even know how often that condition will be hit? I'd make it time based.

**Review by elsirion — APPROVED** [review]
Great optimizations! :rocket:

File: fedimint-rocksdb/src/lib.rs (hunk 1 above)
Comment (dpc) [reply fedimint-rocksdb/src/lib.rs:165]: Yeah, this one is purely a hack. I might need to clean it up into a proper abstraction.

**Comment by dpc** [comment]
Turns out all these optimization don't matter AFAICT, and actually just switching to jemalloc behaves better. @elsirion

**Pull request closed without merging**